R/maths_correctForMemoryEffect.R

```diff
@@ -46,6 +46,30 @@ calculateMemoryCoefficients <- function(dataset) {
   # join the mean mem coeff and the mem coeff for each standard into one dataframe
   tablesToJoin <- append(memoryCoeffForEachStandard, list(meanMemoryCoefficients))
   memCoeffOutput <- reduce(tablesToJoin, full_join, by = "Inj Nr")
+
+  # remove all trailing rows from output dataframe which only contain NA values
+  memCoeffOutput <- memCoeffOutput %>%
+    filter(cumall(!is.na(memoryCoeffD18O)))
+
+  # pad mean memory coefficients with 1's if last coefficient refers to an
+  # injection number less than the maximum inj. number of the samples/standards
+  # in the other blocks
+
+  # bypass if only block 1 is passed as dataset
+  if (nrow(dataset) > nrow(filter(dataset, block == 1))) {
```

Comment:
You could simplify the condition to `any(dataset$block != 1)`

Reply:
This part is no longer needed, but let me comment that I think that this condition would throw an error on a block column like `c(1, 1, 1, NA, NA, NA, ...)`, i.e. just a first block of standards in the measurement file.